Fix not expanded problem when passing an Array object as argument to the where method using composed_of column.

orekyuu opened the pull request:
Fixes #31723

```
david_balance = customers(:david).balance
Customer.where(balance: [david_balance]).to_sql

# Before: WHERE `customers`.`balance` = NULL
# After : WHERE `customers`.`balance` = 50
```

kamipo commented:
Can you more exercise tests for `customer.address` (mapping size: 2)?
Looks like this only work in case of mapping size: 1.

orekyuu commented:
@kamipo Added test with multiple mappings.

kamipo commented:
Originally, if mapping size is 1, `Customer.where(fullname: customer.fullname.to_s)` treated as `Customer.where(name: customer.fullname.to_s)`. If mapping size is more than 1, not allowed as invalid value.

In this PR, `Customer.where(address: customer.address.country)` will be treated as `Customer.where(address_street: customer.address.country, address_city: customer.address.country, address_country: customer.address.country)`.
At least it should not be changed to fix #31723.

Commit:
Fix not expanded problem when passing an Array object as argument to the where method using composed_of column.

Fixes #31723

```
david_balance = customers(:david).balance
Customer.where(balance: [david_balance]).to_sql

# Before: WHERE `customers`.`balance` = NULL
# After : WHERE `customers`.`balance` = 50
```

kamipo committed:
Allow expanding an array of `composed_of` objects
